icu4c/source/common/dictbe.cpp

```diff
@@ -1361,16 +1389,22 @@ CjkBreakEngine::divideUpDictionaryRange( UText *inText,
     // while reversing t_boundary and pushing values to foundBreaks.
     int32_t prevCPPos = -1;
     int32_t prevUTextPos = -1;
-    int correctedNumBreaks = 0;
-    for (int32_t i = numBreaks-1; i >= 0; i--) {
+    int32_t correctedNumBreaks = 0;
+    for (int32_t i = numBreaks - 1; i >= 0; i--) {
         int32_t cpPos = t_boundary.elementAti(i);
         U_ASSERT(cpPos > prevCPPos);
         int32_t utextPos =  inputMap.isValid() ? inputMap->elementAti(cpPos) : cpPos + rangeStart;
         U_ASSERT(utextPos >= prevUTextPos);
         if (utextPos > prevUTextPos) {
             // Boundaries are added to foundBreaks output in ascending order.
             U_ASSERT(foundBreaks.size() == 0 || foundBreaks.peeki() < utextPos);
-            if (!(foundBreaks.contains(utextPos) || utextPos == rangeStart)) {
+            // In phrase breaking, there has to be a breakpoint between Cj character and close
+            // punctuation.
+            // E.g.［携帯電話］正しい選択 -> ［携帯▁電話］▁正しい▁選択 -> breakpoint between ］ and 正
+            if (!foundBreaks.contains(utextPos)
+                && (utextPos != rangeStart
+                       || (isPhraseBreaking && utextPos == rangeStart && utextPos > 0
```

Comment:
No need for "&& utextPos == rangeStart" here because if  utextPos != rangeStart then it won't get into the part after ||.   so you only need
```
if (!foundBreaks.contains(utextPos)
                && (utextPos != rangeStart
                       || (isPhraseBreaking && utextPos > 0
                              && fClosePunctuationSet.contains(utext_char32At(inText, utextPos - 1))))) {

Reply:
Resolved